miner/payload_building.go

```diff
@@ -191,12 +213,14 @@ func (payload *Payload) ResolveFull() *engine.ExecutionPayloadEnvelope {
 	return engine.BlockToExecutableData(payload.full, payload.fullFees, payload.sidecars)
 }
 
-// buildPayload builds the payload according to the provided parameters.
-func (w *worker) buildPayload(args *BuildPayloadArgs) (*Payload, error) {
+func (payload *Payload) interruptWorker() {
+	payload.interrupt.Store(commitInterruptResolve)
+}
+
+func (w *worker) emptyBlock(args *BuildPayloadArgs) *newPayloadResult {
```

Comment:
I might put this back into `buildPayload` if we don't need it in the payload updating goroutine. Initially moved it here into its own function to avoid code duplication.